RELEASING.adoc

```diff
@@ -1,80 +1,177 @@
-```
+Releasing Kiali UI should come before releasing the core, as the image is
+created as soon as the tag for core is pushed to master.
+
+== Publish API docs
 
-Note that this is the same version string that will also tag the kiali repo - this is to ensure they are in-sync and can be used to link the kiali and kiali-ui repo at the time of the release. In other words, this is not the kiali-ui version string that we are getting in the next step.
+API docs are generated by swagger, and this comment will both install it and
+generate the updated documentation.
 
-2) Go to https://www.npmjs.com/package/@kiali/kiali-ui and find the version of the UI that should be built for this release. We will put this console version number in Kiali's Makefile - see below.
+[source, bash]
+----
+$ make swagger-install swagger-gen
+----
 
-_Finally, it is still undetermined how future releases are going to be done with respect to kiali-ui and its own version string. It is possible kiali-ui and kiali repos will be combined into one and thus both kiali and the kiali-ui version strings will be united as a single version string. Today, the version string of kiali-ui as it is seen in the NPM repository is determined within the kiali-ui travis configuration file. This may change in the future. But for now, we need to go to the NPM repository to get the version string of kiali-ui that is to be pulled into the kiali image._
+== Update Makefile
 
-== Kiali
+Change the VERSION variable on Makefile to the new version, this is what is
+going to be shown on the UI, and also what Travis will use to trigger the
+release process.
 
-1) Update the link:Makefile[]
+[source, diff]
+----
+diff --git a/Makefile b/Makefile
+index c1f7f80..aa0c54a 100644
+--- a/Makefile
++++ b/Makefile
+@@ -3,7 +3,7 @@ SHELL=/bin/bash
 
-* The VERSION variable should be set to the new version string of the upcoming release.
-* The CONSOLE_VERSION variable should be set to the kiali-ui npm version string corresponding to the console version to be used.
+ # Identifies the current build.
+ # These will be embedded in the app and displayed when it starts.
+-VERSION ?= v0.5.1-SNAPSHOT
++VERSION ?= v1.0.0
+ COMMIT_HASH ?= $(shell git rev-parse HEAD)
 
-For example,
+ # Indicates which version of the UI console is to be embedded
+----
 
-```
-VERSION ?= v1.0.0
-...
-CONSOLE_VERSION ?= 0.2.11
-```
+By default, the Makefile gets the latest version of the UI, so we'll also
+change the console version on the Makefile:
 
-2) Commit that change to your private GitHub repository:
+[source, diff]
+----
+diff --git a/Makefile b/Makefile
+index c1f7f80..1d05ff3 100644
+--- a/Makefile
++++ b/Makefile
+@@ -13,7 +13,7 @@ COMMIT_HASH ?= $(shell git rev-parse HEAD)
+ # WARNING: If you have previously run the 'docker' target but
+ # later want to change the CONSOLE_VERSION then you must run
+ # the 'clean' target first before re-running the 'docker' target.
+-CONSOLE_VERSION ?= latest
++CONSOLE_VERSION ?= v1.0.0
```

Comment:
Today, the  CONSOLE_VERSION is very different than the VERSION. We should tell people this CONSOLE_VERSION comes from the npm repository and may not be (in fact, today it is not) the same as VERSION. This is the npm repo link where you get the console version:

https://www.npmjs.com/package/@kiali/kiali-ui?activeTab=versions

Reply:
True, added an addendum on it.